README.md

```diff
@@ -52,32 +55,28 @@ replaces the setting of the same name if it already exists. If you want to
 remove a specific setting, use `state: absent` instead of giving a `value`. If
 you want to remove all of the existing `sysfs` settings and replace them with
 the given settings, specify `previous: replaced` as one of the values in the
-list.  If you want to remove all of the `sysfs` settings, use the `dict` value `{"state": "empty"}`, instead of a `list`,
-as the only value for the parameter. See below for examples.
-
-`kernel_settings_systemd_cpu_affinity` - A space delimited list of cpu numbers.
-See https://www.freedesktop.org/software/systemd/man/systemd-system.conf.html#CPUAffinity=
-
-`kernel_settings_systemd_cpu_affinity_state` - Set this to `absent` to remove
-the setting `kernel_settings_systemd_cpu_affinity`, which will be reverted
-back to the system default value upon reboot.
-
-`kernel_settings_transparent_hugepages` - One of the following values:
-`always` `madvise` `never`. This is the memory subsystem transparent hugepages
-value.
-
-`kernel_settings_transparent_hugepages_state` - Set this to `absent` to remove
-the setting `kernel_settings_transparent_hugepages`, which will be reverted
-back to the system default value upon reboot.
-
-`kernel_settings_transparent_hugepages_defrag` - One of these values: `always`
-`defer` `defer+madvise` `madvise` `never`. This is the memory subsystem
-transparent hugepages fragmentation handling value.  The actual supported
-values may be different depending on your OS.
+list.  If you want to remove all of the `sysfs` settings, use the `dict` value
+`{"state": "empty"}`, instead of a `list`, as the only value for the parameter.
+See below for examples.
 
-`kernel_settings_transparent_hugepages_defrag_state` - Set this to `absent` to
-remove the setting `kernel_settings_transparent_hugepages_defrag`, which will
-be reverted back to the system default value upon reboot.
+`kernel_settings_systemd_cpu_affinity` - To set the value, specify a `string`
+which is a space delimited list of cpu numbers. See
+https://www.freedesktop.org/software/systemd/man/systemd-system.conf.html#CPUAffinity=
+If you want to remove the setting, use the `dict` value `{"state": "empty"}`,
+instead of a `string`, as the value for the parameter.
+
+`kernel_settings_transparent_hugepages` - To set the value, specify one of the
+following `string` values: `always` `madvise` `never`. This is the memory
+subsystem transparent hugepages value.  If you want to remove the setting, use
+the `dict` value `{"state": "empty"}`, instead of a `string`, as the value for
+the parameter.
+
+`kernel_settings_transparent_hugepages_defrag` - To set the value, specify one
+of the following `string` values: `always` `defer` `defer+madvise` `madvise`
+`never`. This is the memory subsystem transparent hugepages fragmentation
+handling value.  The actual supported values may be different depending on your
+OS.  If you want to remove the setting, use the `dict` value
+`{"state": "empty"}`, instead of a `string`, as the value for the parameter.
```

Comment:
shouldn't it be `{"state": "absent"}` ? `{"state": "empty"}` is for setting a list an empty list, while `{"state": "absent"}` is for removing it completely.